src/main/java/seedu/address/logic/commands/ClientNoteDeleteCommand.java

```diff
@@ -44,21 +44,17 @@ public ClientNoteDeleteCommand(Index targetClientIndex, Index targetClientNoteIn
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Client> lastShownClientList = model.getSortedFilteredClientList();
-        List<Note> lastShownClientNoteList = model.getSortedFilteredClientNotesList();
-
         if (targetClientIndex.getZeroBased() >= lastShownClientList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_CLIENT_DISPLAYED_INDEX);
         }
-
-        if (targetClientNoteIndex.getZeroBased() >= lastShownClientNoteList.size()) {
+        List<Note> notesList = lastShownClientList.get(targetClientIndex.getZeroBased()).getClientNotesAsList();
+        if (targetClientNoteIndex.getZeroBased() >= notesList.size()) {
             throw new CommandException(MESSAGE_INVALID_CLIENT_NOTE_DISPLAYED_INDEX);
         }
```

Comment:
If the note index is invalid, the message shown is "The client index provided is invalid".